torch/csrc/distributed/c10d/socket.cpp

```diff
@@ -182,7 +182,7 @@ class SocketImpl {
     return hnd_;
   }
 
-  const std::optional<::addrinfo>& remote() const noexcept {
+  const std::optional<std::string>& remote() const noexcept {
```

Comment:
I don't see any changes to any callers, won't it break them?

Reply:
@kurman we only use this for logging -- so it was only passed into fmt::format directly. I'll clean the caller side up since format is no longer necessary